packages/formation-react/src/components/modal/Modal.jsx

```diff
@@ -41,14 +41,18 @@ class Modal extends React.Component {
 
   teardownModal() {
     if (this.state.lastFocus) {
-      this.state.lastFocus.focus();
+      // Ensure last focus is set before completing modal teardown
+      setTimeout(() => {
+        this.state.lastFocus.focus();
+      }, 0)
     }
     document.body.classList.remove('modal-open');
     document.removeEventListener('keydown', this.handleDocumentKeyDown, false);
     document.removeEventListener('focus', this.handleDocumentFocus, true);
     if (this.props.clickToClose) {
       document.removeEventListener('click', this.handleDocumentClicked, true);
     }
+    this.setState({ lastFocus: null })
```

Comment:
this line needs to be moved into the timeout callback as well or it creates a bug with management of the buttons that launch the DD modal

Reply:
but that seems to lead to a react error indicating a memory leak. looking into a better fix.

Reply:
@erikphansen fixed

Reply:
@erikphansen just pushed a change that seems to solve for all.. we only clear state on the last step of the component unmounting